common/common.go

```diff
@@ -486,3 +487,21 @@ func GetExitStatus(err error) (int, error) {
 	}
 	return -1, err
 }
+
+// GetArch returns the current architecture. It returns runtime.GOARCH if not on a arm device
+// and armv6l, armv7l, arm64 otherwise
+func GetArch() string {
+	arch := runtime.GOARCH
+	if arch == "arm" {
+		utsName := &syscall.Utsname{}
+		syscall.Uname(utsName)
+		arch = ""
+		for _, b := range utsName.Machine {
+			if b == 0 {
+				break
+			}
+			arch += string(b)
+		}
+	}
+	return arch
```

Comment:
This information should be available at runtime as the `runtime.goarm` symbol, but it is private. I don't have any ARM device to test, but can you quickly check if something like https://gist.github.com/lucab/f7162ca2d95191c692edc12ea8ccaaef works to get the value?